streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamImpl.java

```diff
@@ -345,6 +347,12 @@ public void writeAsText(final String filePath,
         return branchChildren;
     }
 
+    
+    @Override 
+    public KStream<K, V> merge(KStream<K, V> stream) {
+        return this.builder.merge((KStream<K, V>) this, stream);
```

Comment:
This seems to be a call to the internal implementation of the deprecated method... We should not use this anymore but remove from internal implementation to cleanup the code.